projects/cds/src/merchandising/facade/cds-merchandising-product.service.ts

```diff
@@ -18,31 +33,74 @@ export class CdsMerchandisingProductService {
     protected strategyConnector: MerchandisingStrategyConnector,
     protected baseSiteService: BaseSiteService,
     protected languageService: LanguageService,
-    private routingService: RoutingService
+    private routingService: RoutingService,
+    protected currencyService: CurrencyService,
+    private productSearchService: ProductSearchService,
+    private converterService: ConverterService
   ) {}
 
+  protected readonly RELEVANCE_CATEGORY = ':relevance:category:';
+  protected readonly RELEVANCE_BRAND = ':relevance:brand:';
+  protected defaultPageSize = 10;
+
   loadProductsForStrategy(
     strategyId: string,
     numberToDisplay?: number
   ): Observable<MerchandisingProducts> {
     return combineLatest([
       this.baseSiteService.getActive(),
       this.languageService.getActive(),
-      this.routingService
-        .getRouterState()
-        .pipe(map(state => state.state.params['productCode'])),
+      this.routingService.getRouterState(),
+      this.getFacets(),
     ]).pipe(
-      map(([site, language, productId]: [string, string, string]) => {
-        const strategyRequest: StrategyRequest = {
-          site,
-          language,
-          pageSize: numberToDisplay,
-          productId,
-        };
-        return strategyRequest;
-      }),
-      switchMap(context =>
-        this.strategyConnector.loadProductsForStrategy(strategyId, context)
+      map(
+        ([site, language, routerState, facets]: [
+          string,
+          string,
+          RouterState,
+          string
+        ]) => {
+          const productId = routerState.state.params['productCode'];
+          const category = routerState.state.params['categoryCode'];
+          const strategyRequest: StrategyRequest = {
+            site,
+            language,
+            pageSize: numberToDisplay,
+            productId,
+            category,
+            facets,
+          };
+          return strategyRequest;
+        }
+      ),
+      tap(request => console.log('the request we are using - ', request)),
+      distinctUntilChanged(
```

Comment:
Is this really necessary? Something (`private getFacets()`?) is firing too many times?

If this is the case, maybe you can add `.pipe(distinctUntilChanged())` to each of the combined streams. `distinctUntilChanged()` works best when it operates with a simple type - `string`, `number`, `boolean`, etc. Maybe that can be leveraged when deciding how to split into smaller streams.

If everything works out great, then I'm hoping we can remove this complex `distingUntilChanged()` and effectively remove `projects/cds/src/utils/compare-equals-objects.ts`

Reply:
This was put in because when you click on a facet (lets say starty with the URL https://localhost:4200/electronics-spa/en/USD/Open-Catalogue/Cameras/Film-Cameras/c/574?query=:relevance:category:574:price:$1,000-$100,000:price:$0-$49.99) to either add/remove a facet for searching, the routerService emits another item, as well as the productSearchService, so it tries to make another two calls to this.strategyConnector.loadProductsForStrategy, which then get immediately cancelled.

Reply:
This is already changed, but just to follow up:
`switchMap` does the cancelling of the current ongoing request, and switches immediately to the newly emitted value.

What do you want to do in this case? You have a couple of options:
- queue the emitted values, and then call the API for each of them? `concatMap`
- ignore the new value if the API is currently being called - `exhaustMap`
- cancel the current API and jump on the newly emitted value - `switchMap`
- do a new API call in parallel - `mergeMap`

Reply:
I've currently changed this to a flatMap

Reply:
`flatMap` is an alias for `megeMap`, so the later should be used.